packages/clay-datepicker/src/DateNavigation.js

```diff
@@ -0,0 +1,153 @@
+import * as Helpers from './Helpers';
+import moment from 'moment';
+import PropTypes from 'prop-types';
+import React, {useMemo} from 'react';
+
+import Button from './Button';
+import Select from './Select';
+
+/**
+ * @param {Object} param
+ * @return {React.createElement}
+ */
+function DateNavigation({
+	currentMonth,
+	months,
+	onDotClicked,
+	onMonthChange,
+	spritemap,
+	years,
+}) {
+	const memoizedYears = useMemo(
+		() =>
+			Helpers.range(years).map(elem => {
+				return {
+					label: elem,
+					value: elem,
+				};
+			}),
+		[years.start, years.end]
+	);
+
+	const memoizedMonths = useMemo(
+		() =>
+			months.map((month, index) => {
+				return {
+					label: month,
+					value: index,
+				};
+			}),
+		months
+	);
+
+	/**
+	 * Handles the change of the month from the available
+	 * years in the range
+	 * @param {!number} month
+	 */
+	function handleChangeMonth(month) {
+		const date = moment(currentMonth)
+			.clone()
+			.add(month, 'M')
+			.toDate();
+
+		if (memoizedYears.find(elem => elem.value === date.getFullYear())) {
```

Comment:
Maybe some harmless micro-optimization to do here and pull the `getFullYear()` outside of the `find`? ie.

```javascript
const year = date.getFullYear();
```

Reply:
Oh yeah, thanks!